src/js/bsp-modal.js

```diff
@@ -135,17 +128,22 @@ var bsp_modal = {
     // private function to open modal from the DOM. Calls the vex open method, but also sets up
     // the centering and passes though the events. Lastly, upon close, it puts everything back the
     // way it founded, leaving the DOM intact
-    _openFromDOM: function(options) {
+    _openFromDOM: function(opener) {
         var self = this;
-        // just in case no options are passed in
-        options = options || {};
         // grab the modal data within that element
         var $modalData = self.$el.find('.modal-data').contents();
         // save it off, we're going to need to put it back
         var $savedContents = $modalData.clone();
 
-        if(options.hash) {
-            window.location.hash = options.hash;
+        // if opened by a DOM element
+        if(opener) {
+            var $opener = $(opener);
+            var href = $opener.attr('href');
+            // if we have a anchor link, let that through
+            if(href && href.indexOf('#') === 0) {
+                window.location.hash = href;
+            }
```

Comment:
The old conditional remains unchanged except for a test of the existence of "href" (previously you couldn't open a modal from anything but a link because "href" would not have the indexOf() method and an error would be thrown).